dao/src/main/scala/za/co/absa/enceladus/dao/rest/RestClient.scala

```diff
@@ -85,7 +85,7 @@ protected class RestClient(authClient: AuthClient,
         log.warn(s"Response - $statusCode : ${Option(response.getBody).getOrElse("None")}")
         log.warn(s"Unauthorized $method request for Menas URL: $url")
         if (retriesLeft <= 0) {
-          throw UnauthorizedException("Unable to reauthenticate, no retries left")
+          throw OptionallyRetryableException.UnauthorizedException("Unable to reauthenticate, no retries left")
```

Comment:
I still haven't split this case `case HttpStatus.UNAUTHORIZED | HttpStatus.FORBIDDEN =>` here into two cases. I started but then I realized that if this is split into two exceptions and two case branches, then exactly the same code would be executed (doesn't matter if this is encapsulated into a sub-function or not, that's just a cosmetic thing). 

`UnauthorizedException` is used in a couple of places and the usage is quite straightforward - this will be raised/caught whenever there are problems with authorization. So I've decided to keep it as is for now. But I'm happy to be persuaded.

Reply:
This is the last remaining open item, I see. Let's bring it up at tomorrow stand-up.